File: net.nestjs.kafka/src/utils.common/BaseResponse.ts
import { HttpStatus } from "@nestjs/common";

export class BaseResponse {
    
    private status: HttpStatus;
    private message: string;
    private data: Object

    constructor(status: number = 0, message: string = 'NULL', data?: Object) {
        this.status = status ? + status : + HttpStatus.OK;
        this.message = message ? message : "SUCCESS";
        this.data = data ? data : {};
    }

    
    public getStatus(): HttpStatus {
        return this.status;
    }

    public setStatus(status: HttpStatus): void {
        this.status = status;
    }

    public getMessage(): string {
        return this.message;
    }

    public setMessage(status: HttpStatus, message: string): void {

        if (message) {
            this.message = message;
        } else {
            switch (status) {
                case HttpStatus.OK:
                    this.message = "SUCCESS";
                    break;
                case HttpStatus.BAD_REQUEST:
                    this.message = "Dữ liệu không hợp lệ";
                    break;
                default:
                    this.message = "SUCCESS";
                    break;
            }
        }
    }

    public getData(): Object {
        return this.data;
    }

    public setData(data: Object): void {
        this.data = data;
    }
}

File: net.java.kafka.producer/src/main/java/net/java/kafka/producer/response/BaseResponse.java
package net.java.kafka.producer.response;

import org.springframework.http.HttpStatus;

public class BaseResponse<T> {

	private int status;
	private String message;
	private T data;

	public BaseResponse() {
		this.setStatus(HttpStatus.OK);
		this.setMessage(HttpStatus.OK);
		this.setData(null);
	}

	public int getStatus() {
		return status;
	}

	public void setStatus(HttpStatus statusEnum) {
		this.status = statusEnum.value();
		this.message = statusEnum.name();
	}

	public String getMessage() {
		return this.message;
	}

	public void setMessage(HttpStatus statusEnum) {
		this.message = statusEnum.name();
	}

	public void setMessageError(String message) {
		this.message = message;
	}

	public T getData() {
		return data;
	}

	public void setData(T data) {
		this.data = data;
	}

}
